### ISSUE TITLE
Hardware checksum offloading toggles only IPv4 not IPv6 related setting

### ISSUE THREAD

**Issue by kevemueller** [opnsense/core#1227]
The Interfaces: Settings/Hardware CRC toggle switches on/off the txcsum and rxcsum interface options, but does not touch the rxcsum6, txcsum6 options.
The way this option is phrased in the UI, one would expect it to switch on/off both IPv4 as well  as IPv6 hardware checksum offloading.
Ideally these settings should be split to the granularity of the underlying ifconfig, i.e. (tx|rx)csum6?, tso(4|6), lro, vlanhw(csum|tso).
This might be tackled together with #1137 .

**Comment by AdSchellevis** [comment]
@kevemueller I'm not sure if @pv2b is still working on this, but these tags live on a different place in the code.
https://github.com/opnsense/core/blob/master/src/etc/inc/interfaces.lib.inc#L236-L271

Hardware offloading in FreeBSD is quite sensitive and causes a lot of issues, that's why we choose to set the defaults to disabled some time ago https://github.com/opnsense/core/issues/1092 .
The mechanism in configure_interface_hardware() checks the settings already on the card before setting them, because in some scenarios even the network link gets lost when "changing" offloading settings to the value they already had. 

I didn't change the vlan settings at the time I fixed the other offloading settings https://github.com/opnsense/core/commit/2149c8972c7d6a293ce0668d7ffa408bd7ddde46 but it probably would have been better to move those in there as well using the same strategy. Testing just costs a lot of time, which just doesn't fit my schedule at the moment.

Long story short, adding rxcsum6/txcsum6 to configure_interface_hardware() would be easy and should in my opinion be toggled by "Disable hardware checksum offload", which also aligns with the other settings in the interface settings (on/off, means both ipv4/ipv6).
Moving the vlan tags is something for another day, mainly for testing reasons. 

If I add the rxcsum6, txcsum6 options, do you have time to test? (should have little impact)

**Comment by kevemueller** [comment]
Yes, checksum offloading hit me hard in a VMXNET3 UDP6 environment. I remembered that it was disabled on UI, but the details are in the problem report. I am happy you agree with the toggle to apply to IPv6 settings as well.
I fixed my environment by manually disabling offloading. I believe the problem will reappear after reboot - which means I should be able to test impact of a patch.
I am happy to support with the testing part.

**Comment by AdSchellevis** [comment]
@kevemueller can you try https://github.com/opnsense/core/commit/a801eb11bd1c47e95a4318f413cc291f512d4b7e ?

`opnsense-patch  a801eb11bd
`

**Comment by fichtner** [comment]
@kevemueller @AdSchellevis  the commit is on 16.7.7-devel, but I'm also merging this for 16.7.8. Looks good so far. Closing this for now unless there is more issues with it...
